protocols/kad/src/behaviour.rs

```diff
@@ -283,6 +316,26 @@ impl<TSubstream> Kademlia<TSubstream> {
         self.start_query(QueryInfoInner::GetProviders { target, pending_results: Vec::new() });
     }
 
+    /// Starts an iterative `GET_VALUE` request.
+    pub fn get_value(&mut self, key: &Multihash) {
+        if let Some(value) = self.records.get(key) {
+            self.queued_events.push(NetworkBehaviourAction::GenerateEvent(
+                KademliaOut::GetValueResult {
+                    result: Some((key.clone(), value.clone())),
+                    closer_peers: None,
+                }
+            ));
+        }
+        self.start_query(QueryInfoInner::GetValue { key: key.clone(), results: vec![] });
+    }
+
+    /// Starts an iterative `PUT_VALUE` request
+    pub fn put_value(&mut self, key: Multihash, value: Vec<u8>) {
+        // TODO: Probably we shouldn't store the value ourselves
```

Comment:
Right, this really shouldn't be here. The local node should only store the value if it is itself among the `k` closest to the key and it can only know that once the query is complete.